include/aspect/simulator.h

```diff
@@ -594,6 +594,8 @@ namespace aspect
       void
       local_assemble_advection_system (const AdvectionField &advection_field,
                                        const std::pair<double,double> global_field_range,
+                                       const bool use_smoothing, \
+                                       Vector<float> viscosity_per_cell, \
```

Comment:
You definitely don't want to pass this vector by copy. It should certainly be a reference, and probably a const reference.

Out of curiosity, why do you let this function do the averaging, rather than have a separate function takes this vector and already does all of the averaging for you, before you even pass it here?